### PR TITLE
[libc][math] Add MPFR unit tests for nearbyint{,f,l,f16}

### PR DESCRIPTION
**Pull request by overmighty**

### CODE REVIEW

**Comment by overmighty** [comment]
cc @lntue

Based on #94473.

File: libc/test/src/math/RIntTest.h [hunk 1]
```diff
@@ -131,7 +138,7 @@ class RIntTestTemplate : public LIBC_NAMESPACE::testing::FEnvSafeTest {
       for (int mode : ROUNDING_MODES) {
         LIBC_NAMESPACE::fputil::set_round(mode);
         mpfr::RoundingMode mpfr_mode = to_mpfr_rounding_mode(mode);
-        ASSERT_FP_EQ(func(x), mpfr::round(x, mpfr_mode));
+        EXPECT_FP_EQ(func(x), mpfr::round(x, mpfr_mode));
```
Comment (overmighty) [inline libc/test/src/math/RIntTest.h:141]: Would it be worth it to also use `EXPECT_FP_EQ_WITH_EXCEPTION` here based on the mantissa of `x`/`xbits`?

**Comment by overmighty** [comment]
Marking as draft until #94383 is relanded.

**Comment by overmighty** [comment]
cc @lntue

File: libc/test/src/math/NearbyIntTest.h [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+#include "hdr/fenv_macros.h"
+
+namespace mpfr = LIBC_NAMESPACE::testing::mpfr;
+
+template <typename T>
+class NearbyIntTestTemplate : public LIBC_NAMESPACE::testing::FEnvSafeTest {
+
+  DECLARE_SPECIAL_CONSTANTS(T)
+
+  static constexpr int ROUNDING_MODES[4] = {FE_UPWARD, FE_DOWNWARD,
```
Comment (lntue) [inline libc/test/src/math/NearbyIntTest.h:28]: Why don't you use:
```
using namespace LIBC_NAMESPACE::fputil::testing;
static constexpr RoundingMode ROUNDING_MODES[4] = { RoundingMode::Upward, ... };
```

Then you can replace all the `LIBC_NAMESPACE::fputil::set_round()` below with:
```
    ForceRoundingMode __r(mode);
    if (__r.success) {
      ...
    }
```

Comment (overmighty) [reply libc/test/src/math/NearbyIntTest.h:28]: I copied RIntTest.h and did a bit of cleanup but not that much. Will change this.

**Review by lntue — APPROVED** [review]

**Pull request merged**